File: Notificare.Push.iOS.Binding.Bridge/NotificarePushBinding/Sources/Models/NotificareNotificationDeliveryMechanism.swift
import Foundation
import NotificarePushKit

@objc
public enum NotificareNotificationDeliveryMechanism : Int {
    case standard
    case silent

    init(from deliveryMechanism: NotificarePushKit.NotificareNotificationDeliveryMechanism) {
        switch deliveryMechanism {
        case .standard:
            self = .standard
        case .silent:
            self = .silent
        @unknown default:
            fatalError("Unknown delivery mechanism: \(deliveryMechanism.rawValue)")
        }
    }
}

File: Notificare.Push.Core/Models/NotificareNotificationDeliveryMechanism.cs
namespace NotificareSdk.Push.Core.Models;

public enum NotificareNotificationDeliveryMechanism
{
    Standard,
    Silent,
}

public static class NotificareNotificationDeliveryMechanismExtensions
{
    public static string ToRawValue(this NotificareNotificationDeliveryMechanism deliveryMechanism)
    {
        return deliveryMechanism switch
        {
            NotificareNotificationDeliveryMechanism.Standard => "standard",
            NotificareNotificationDeliveryMechanism.Silent => "silent",
            _ => throw new ArgumentException(
                $"Unknown {nameof(NotificareNotificationDeliveryMechanism)}: {deliveryMechanism}"
            )
        };
    }
}
